cd/auto-generate/auto-tag-controller.sh

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+USAGE="
+Usage:
+  $(basename "$0")
+
+This script automatically tags the service controller repository with proper semver
+to automate the controller release process.
+The script currently only support automating patch releases.
+(TODO: Consider automating Minor and Major release for controllers as well)
+
+The script compares latest GitHub tag on service controller repository with the
+image tag inside helm artifacts. If the generated helm artifacts are valid next
+patch release, then this scripts tags the service controller repository with
+helm image tag version.
+
+This tagging along with GitHub action for creating GitHub
```

Comment:
GitHub Action for creating automating release (whenever a semver tag is added) is very clean, efficient and requires no code on our part. 
I am happy with just sticking to the release workflow from GitHub actions. [Sample release](https://github.com/aws-controllers-k8s/elasticache-controller/releases/tag/v0.0.6)

Reply:
I have corrected the documentation now. 
When we push new git tag, the release is automatically created. I just tested it.
The github action workflow just updates the existing release with content like commits and contributors etc.

So having GHAction WF  is not needed for this automation, I will still add this in initial onboarding guidance so that our patch releases have commit descriptions.

Reply:
There are tools that generate a nice changelog from commits, that can easily replace our GH workflow. Sure it doesn't need to be a task for this PR, but I believe moving to that will actually be better. Although the GH workflow was fun, managing it across every repository is a PITA.

Reply:
> Although the GH workflow was fun, managing it across every repository is a PITA.

IMO, it is not a major pain. It's one time setup just like adding some initial files to new controller repo and then just leave it be for rest of the time. Having it in `.github` also makes the setup super easy